### ISSUE TITLE
Additional functionalities for basic training with tfe.keras

### ISSUE THREAD

**Issue by yanndupis** [tf-encrypted/tf-encrypted#640]
## Context 

We are now able to support simple models (e.g logistic regression) with tfe.keras #606. See [this notebook](https://github.com/tf-encrypted/tf-encrypted/blob/master/examples/notebooks/keras-training/Logistic%20Regression.ipynb) as an example. 

We would like to add the following basic functionalities:
- Add accuracy metric to [compile method](https://www.tensorflow.org/api_docs/python/tf/keras/Sequential#compile).
- Add the option to select optimizer using string in [compile](https://www.tensorflow.org/api_docs/python/tf/keras/Sequential#compile).
- [fit](https://www.tensorflow.org/api_docs/python/tf/keras/Sequential#fit) method currently expect private tensors for input `x` and `y` but we should support numpy arrays as an input.
- Add to BinaryCrossentopy [from_logits](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/BinaryCrossentropy#__init__) support.
- Create TFE version of [tf.data.Dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#class_dataset) which will be used as `model.fit()` input like in normal Keras.
